### PR TITLE
Internal text field

### PR DESCRIPTION
**Pull request by MHKalantarian**
I have extracted a widget to type inside the keyboard layout
Problems are:
Backspace button not deleting anything
Focus not lost when touched outside keyboard

#1888

### CODE REVIEW

**Comment by patrickgold** [comment]
Thanks a lot for your PR, really appreciate it!

I've had a quick look at the changes and the base idea is good!

The main problem with focus on Android is that the concept is completely stupid and does not work in anyway that would make sense, JC makes that even worse. I will have a look later today and play around with your code to address your mentioned problems (have some idea but am not sure if it is the cause), if I have results then I will post the patch commit here.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
+#Tue May 31 12:14:01 IRDT 2022
 distributionBase=GRADLE_USER_HOME
-distributionPath=wrapper/dists
-distributionSha256Sum=29e49b10984e585d8118b7d0bc452f944e386458df27371b49b4ac1dec4b7fda
 distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-bin.zip
-zipStoreBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
 zipStorePath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
```
Comment (patrickgold) [inline gradle/wrapper/gradle-wrapper.properties:6]: In the meantime, could you please revert the changes of the gradle-wrapper.properties file please? This file is auto-generated by my Gradle setup and thus should not be hand-edited. Thanks!

**Review by patrickgold — CHANGES_REQUESTED** [review]

**Comment by patrickgold** [comment]
Sorry for getting back this late, I tried out some changes a week ago but did not succeed and then got distracted. I will currently leave your PR in this state and when I find the time provide a mechanism for the problems you mentioned. I hope that this is okay for you.

**Comment by MHKalantarian** [comment]
Could you point me to the version before you changed everything to compose? I want to try somethings on legacy

**Comment by patrickgold** [comment]
Basically it is the 0.3.13 release tag: https://github.com/florisboard/florisboard/tree/v0.3.13

After that I almost immediately began adding Compose.

**Comment by MHKalantarian** [comment]
I tested out various situations, found a workaround to work with a button instead of handling focus. But the problem with cursor still remains, when typing in internal edit text using it input connection, the cursor does not move forward, therefore backspace won't work and words are backward. Can you suggest anything for this part?

**Comment by MHKalantarian** [comment]
Problem fixed. The only problem that remains is to detect touch gesture outside of keyboard bounds, I have not found any documentation for this. Do you know anything about this?

**Comment by atikattar1104** [comment]
Hi!
Currently Typing With Florisboard.
I Have Tested It. Results As Follows :- 
1) Backspace Is Working In All Edittext Views.
2) Keyboard Is Closing When Touched Outside Keyboard On Webpages And When Touching Clickable Views In Apps (Like, Buttons, Switches, Etc.). But, When You Click On Blank Area (i.e. - Non-Clickable Views) In Apps, Keyboard Doesn't Closes. Same Behaviour Is With Other Keyboards As Well.
Thanks.

**Comment by patrickgold** [comment]
I will close this PR as it is stalled for a long time and not really compatible with the current code base anymore. Adding a text box within the keyboard UI is definitely possible but not easy to get right and then maintain, so for now it is best to just not have it in the keyboard UI. Once the keyboard code base nears 1.0 we can then try adding it in.

**Pull request closed without merging**